nibabel/gifti/parse_gifti_fast.py

```diff
@@ -168,13 +168,14 @@ def StartElementHandler(self, name, attrs):
                 self.da.ind_ord = array_index_order_codes.code[
                     attrs["ArrayIndexingOrder"]]
             if "Dimensionality" in attrs:
-                self.da.num_dim = int(attrs["Dimensionality"])
-            for i in range(self.da.num_dim):
+                self.num_dim = int(attrs["Dimensionality"])
```

Comment:
Oh, I see. Okay, so should you maybe set `self.num_dim = 0` at some point, so that `for i in range(self.num_dim):` doesn't fail if `"Dimensionality" not in attrs`?

Maybe replace the `if` block with:

    self.num_dim = int(attrs.get("Dimensionality", 0))

This is assuming `attrs` has a proper `dict` interface.